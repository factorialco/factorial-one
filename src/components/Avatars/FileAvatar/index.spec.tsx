import { render } from "@testing-library/react-native";
import React from "react";
import { FileAvatar } from "./";

describe("FileAvatar", () => {
  it("Snapshot - different file types", () => {
    const fileTypes = [
      { name: "document.pdf", type: "application/pdf" },
      { name: "image.jpg", type: "image/jpeg" },
      {
        name: "document.docx",
        type: "application/vnd.openxmlformats-officedocument.wordprocessingml.document",
      },
      {
        name: "spreadsheet.xlsx",
        type: "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",
      },
      {
        name: "presentation.pptx",
        type: "application/vnd.openxmlformats-officedocument.presentationml.presentation",
      },
      { name: "text.txt", type: "text/plain" },
      { name: "video.mp4", type: "video/mp4" },
      { name: "audio.mp3", type: "audio/mpeg" },
      { name: "archive.zip", type: "application/zip" },
      { name: "data.csv", type: "csv" },
      { name: "webpage.html", type: "html" },
      { name: "readme.md", type: "markdown" },
      { name: "unknown.xyz", type: "undefined" },
    ];

    fileTypes.map((fileType, index) => {
      const { toJSON } = render(
        <FileAvatar
          key={index}
          file={new File([""], fileType.name, { type: fileType.type })}
        />,
      );

      expect(toJSON()).toMatchSnapshot();
    });
  });
});
