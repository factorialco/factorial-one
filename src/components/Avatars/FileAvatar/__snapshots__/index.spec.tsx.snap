// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileAvatar Snapshot - different file types 1`] = `
<View
  className="inline-flex "
>
  <View
    className="h-fit w-fit"
  >
    <View
      className="flex shrink-0 items-center justify-center overflow-hidden text-center font-semibold w-8 h-8 rounded border border-solid border-f1-border-secondary bg-f1-background"
    >
      <Text
        className="font-semibold text-sm text-f1-foreground-accent"
      >
        PDF
      </Text>
    </View>
  </View>
</View>
`;

exports[`FileAvatar Snapshot - different file types 2`] = `
<View
  className="inline-flex "
>
  <View
    className="h-fit w-fit"
  >
    <View
      className="flex shrink-0 items-center justify-center overflow-hidden text-center font-semibold w-8 h-8 rounded border border-solid border-f1-border-secondary bg-f1-background"
    >
      <Text
        className="font-semibold text-sm text-f1-foreground-info"
      >
        IMG
      </Text>
    </View>
  </View>
</View>
`;

exports[`FileAvatar Snapshot - different file types 3`] = `
<View
  className="inline-flex "
>
  <View
    className="h-fit w-fit"
  >
    <View
      className="flex shrink-0 items-center justify-center overflow-hidden text-center font-semibold w-8 h-8 rounded border border-solid border-f1-border-secondary bg-f1-background"
    >
      <Text
        className="font-semibold text-sm text-f1-foreground-info"
      >
        DOC
      </Text>
    </View>
  </View>
</View>
`;

exports[`FileAvatar Snapshot - different file types 4`] = `
<View
  className="inline-flex "
>
  <View
    className="h-fit w-fit"
  >
    <View
      className="flex shrink-0 items-center justify-center overflow-hidden text-center font-semibold w-8 h-8 rounded border border-solid border-f1-border-secondary bg-f1-background"
    >
      <Text
        className="font-semibold text-sm text-f1-foreground-positive"
      >
        XLS
      </Text>
    </View>
  </View>
</View>
`;

exports[`FileAvatar Snapshot - different file types 5`] = `
<View
  className="inline-flex "
>
  <View
    className="h-fit w-fit"
  >
    <View
      className="flex shrink-0 items-center justify-center overflow-hidden text-center font-semibold w-8 h-8 rounded border border-solid border-f1-border-secondary bg-f1-background"
    >
      <Text
        className="font-semibold text-sm text-f1-foreground-warning"
      >
        PPT
      </Text>
    </View>
  </View>
</View>
`;

exports[`FileAvatar Snapshot - different file types 6`] = `
<View
  className="inline-flex "
>
  <View
    className="h-fit w-fit"
  >
    <View
      className="flex shrink-0 items-center justify-center overflow-hidden text-center font-semibold w-8 h-8 rounded border border-solid border-f1-border-secondary bg-f1-background"
    >
      <Text
        className="font-semibold text-sm text-f1-foreground-secondary"
      >
        TXT
      </Text>
    </View>
  </View>
</View>
`;

exports[`FileAvatar Snapshot - different file types 7`] = `
<View
  className="inline-flex "
>
  <View
    className="h-fit w-fit"
  >
    <View
      className="flex shrink-0 items-center justify-center overflow-hidden text-center font-semibold w-8 h-8 rounded border border-solid border-f1-border-secondary bg-f1-background"
    >
      <Text
        className="font-semibold text-sm text-f1-foreground-info"
      >
        VID
      </Text>
    </View>
  </View>
</View>
`;

exports[`FileAvatar Snapshot - different file types 8`] = `
<View
  className="inline-flex "
>
  <View
    className="h-fit w-fit"
  >
    <View
      className="flex shrink-0 items-center justify-center overflow-hidden text-center font-semibold w-8 h-8 rounded border border-solid border-f1-border-secondary bg-f1-background"
    >
      <Text
        className="font-semibold text-sm text-f1-foreground-accent"
      >
        AUD
      </Text>
    </View>
  </View>
</View>
`;

exports[`FileAvatar Snapshot - different file types 9`] = `
<View
  className="inline-flex "
>
  <View
    className="h-fit w-fit"
  >
    <View
      className="flex shrink-0 items-center justify-center overflow-hidden text-center font-semibold w-8 h-8 rounded border border-solid border-f1-border-secondary bg-f1-background"
    >
      <Text
        className="font-semibold text-sm text-f1-foreground-warning"
      >
        ZIP
      </Text>
    </View>
  </View>
</View>
`;

exports[`FileAvatar Snapshot - different file types 10`] = `
<View
  className="inline-flex "
>
  <View
    className="h-fit w-fit"
  >
    <View
      className="flex shrink-0 items-center justify-center overflow-hidden text-center font-semibold w-8 h-8 rounded border border-solid border-f1-border-secondary bg-f1-background"
    >
      <Text
        className="font-semibold text-sm text-f1-foreground-positive"
      >
        CSV
      </Text>
    </View>
  </View>
</View>
`;

exports[`FileAvatar Snapshot - different file types 11`] = `
<View
  className="inline-flex "
>
  <View
    className="h-fit w-fit"
  >
    <View
      className="flex shrink-0 items-center justify-center overflow-hidden text-center font-semibold w-8 h-8 rounded border border-solid border-f1-border-secondary bg-f1-background"
    >
      <Text
        className="font-semibold text-sm text-f1-foreground-accent"
      >
        HTM
      </Text>
    </View>
  </View>
</View>
`;

exports[`FileAvatar Snapshot - different file types 12`] = `
<View
  className="inline-flex "
>
  <View
    className="h-fit w-fit"
  >
    <View
      className="flex shrink-0 items-center justify-center overflow-hidden text-center font-semibold w-8 h-8 rounded border border-solid border-f1-border-secondary bg-f1-background"
    >
      <Text
        className="font-semibold text-sm text-f1-foreground-secondary"
      >
        MD
      </Text>
    </View>
  </View>
</View>
`;

exports[`FileAvatar Snapshot - different file types 13`] = `
<View
  className="inline-flex "
>
  <View
    className="h-fit w-fit"
  >
    <View
      className="flex shrink-0 items-center justify-center overflow-hidden text-center font-semibold w-8 h-8 rounded border border-solid border-f1-border-secondary bg-f1-background"
    >
      <Text
        className="font-semibold text-sm text-f1-foreground"
      >
        FIL
      </Text>
    </View>
  </View>
</View>
`;
